crates/swc_ecma_transforms_compat/src/es2015/destructuring.rs

```diff
@@ -518,22 +535,27 @@ impl Destructuring {
         }
 
         let stmts = if decls.is_empty() {
-            body.stmts
+            body.stmts.take()
         } else {
-            iter::once(
-                Stmt::Decl(Decl::Var(VarDecl {
-                    span: DUMMY_SP,
-                    kind: VarDeclKind::Let,
-                    decls,
-                    declare: false,
-                }))
-                .fold_with(self),
-            )
-            .chain(body.stmts)
-            .collect()
+            let mut stmt = Stmt::Decl(Decl::Var(VarDecl {
+                span: DUMMY_SP,
+                kind: VarDeclKind::Let,
+                decls,
+                declare: false,
+            }));
+
+            stmt.visit_mut_children_with(self);
```

Comment:
I think this is bit different from original, as original one is `.fold_with()`.
So it should be `visit_mut_with(self)` to be identical.

Reply:
But well, as there's no `visit_mut_stmt` so it does not matter.